parquet-column/src/main/java/org/apache/parquet/column/values/factory/DefaultV2ValuesWriterFactory.java

```diff
@@ -102,19 +125,39 @@ private ValuesWriter getBinaryValuesWriter(ColumnDescriptor path) {
   }
 
   private ValuesWriter getInt32ValuesWriter(ColumnDescriptor path) {
-    ValuesWriter fallbackWriter = new DeltaBinaryPackingValuesWriterForInteger(
-        parquetProperties.getInitialSlabSize(),
-        parquetProperties.getPageSizeThreshold(),
-        parquetProperties.getAllocator());
+    final ValuesWriter fallbackWriter;
+    // Ideally we should only enable BYTE_STREAM_SPLIT if compression is enabled for this column.
+    // However, this information is not available here.
```

Comment:
It is passed down from `ParquetWriter`/`ParquetRecordWriter` so defined per file. I agree it would be nice to have it here but currently I cannot imagine how could we do it.
Feel free to leave this comment here as a TODO, however, since we currently have it explicitly configure by the client, I don't think we need it.

Reply:
Pity. I'll just leave the comment then.